src/modules/ens/sagas.ts

```diff
@@ -497,6 +501,53 @@ export function* ensSaga(builderClient: BuilderClient, ensApi: ENSApi, worldsAPI
     }
   }
 
+  function* handleFetchContributableNamesRequest() {
+    try {
+      let names: ContributableDomain[] = yield call([worldsAPIContent, 'fetchContributableDomains'])
+
+      const bannedNames: string[] = yield call([lists, 'fetchBannedNames'])
+      const bannedNamesSet = new Set(bannedNames.map(x => x.toLowerCase()))
+
+      names = names.filter(({ name }) => name.split('.').every(nameSegment => !bannedNamesSet.has(nameSegment)))
+      const ensDomains: string[] = []
+      const nameDomains: string[] = []
+      names.forEach(({ name }) => {
+        if (name.includes('.dcl.eth')) {
+          nameDomains.push(name.replace('.dcl.eth', ''))
+        } else {
+          ensDomains.push(name)
+        }
+      })
+
+      const [ownerByNameDomain, ownerByEnsDomain]: [Record<string, string>, Record<string, string>] = yield all([
+        call([marketplace, 'fetchENSOwnerByDomain'], nameDomains),
```

Comment:
What do you think about exercising the failure of any of these requests in a test?